src/CodeAnalyzer.py

```diff
@@ -0,0 +1,88 @@
+stack_protectors = ["__stack_chk_fail", "___stack_chk_guard", "in_FS_OFFSET"]
+
+class CodeAnalyzer:
+    def __init__(self, signature, func_code):
+        self.func_code = func_code
+        self.signature = signature
+        self.types_of_variables = {}
+        
+    def get_variable_name(self, variable):
+        if "*" in variable:
+                variable = variable[variable.rfind("*")+1:]
+        if "[" in variable:
+            variable = variable[:variable.find("[")]
+        if ";" in variable:
+            variable = variable[:variable.find(";")]
+        return variable
+    
+    def get_variables_types_from_signature(self):
+        for argument in self.signature[self.signature.find("(")+1:self.signature.find(")")].split(","):
+            if argument == "void":
+                break
+            type, variable = argument.split()
+            variable = self.get_variable_name(variable)
+            self.types_of_variables[variable] = type
+        return self.types_of_variables
+
+    def remove_warnings(self, lines, id, warning_in_load, warning_in_store):
+        lvalue, rvalue = lines[id].split(" = ")
+        variable = self.get_variable_name(lvalue.strip())
+        if warning_in_load:
+            if rvalue[1] == "(":
+                lines[id] = lvalue + " = " + rvalue[0] + f"({self.types_of_variables[variable]} *)" + rvalue[rvalue.index(")")+1:]
+            else:
+                lines[id] = lvalue + " = " + rvalue[0] + f"({self.types_of_variables[variable]} *)" + rvalue[1:]
+        
+        if warning_in_store:
+            if lvalue[1] == "(":
+                lines[id] = lvalue[0] + f"({self.types_of_variables[variable]} *)" + lvalue[lvalue.index(")")+1:] + " = " + rvalue
+            else:
+                lines[id] = lvalue[0] + f"({self.types_of_variables[variable]} *)" + lvalue[1:] + " = " + rvalue
```

Comment:
```suggestion
    def remove_warnings(self, line: str, warning_in_load: bool, warning_in_store: bool) -> str:
        lvalue, rvalue = lines[id].split(" = ")
        variable = self.get_variable_name(lvalue.strip())
        if warning_in_load:
            if rvalue[1] == "(":
                line = lvalue + " = " + rvalue[0] + f"({self.types_of_variables[variable]} *)" + rvalue[rvalue.index(")")+1:]
            else:
                line = lvalue + " = " + rvalue[0] + f"({self.types_of_variables[variable]} *)" + rvalue[1:]
        
        if warning_in_store:
            if lvalue[1] == "(":
                line = lvalue[0] + f"({self.types_of_variables[variable]} *)" + lvalue[lvalue.index(")")+1:] + " = " + rvalue
            else:
                line = lvalue[0] + f"({self.types_of_variables[variable]} *)" + lvalue[1:] + " = " + rvalue
        return line
```
Наверно можно обойтись без lines и id, а просто передавать одну line, и соответственно вызовы remove_warnings(lines, id, warning_in_load, warning_in_store) заменить на lines[id] = remove_warnings(line, warning_in_load, warning_in_store)